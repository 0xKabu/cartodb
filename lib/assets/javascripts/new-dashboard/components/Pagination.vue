<template>
  <ul class="Pagination-list CDB-Text CDB-Size-medium">
    <li class="Pagination-listItem" v-if="showFirst">
      <button class="Pagination-listItemInner Pagination-listItemInner--link" @click=goToPage(1)>1</button>
    </li>
    <li class="Pagination-listItem" v-if="showNPositionPrev(2)">
      <button class="Pagination-listItemInner Pagination-listItemInner--more">&hellip;</button>
    </li>
    <li class="Pagination-listItem" v-if="showNPositionPrev(1)">
      <button class="Pagination-listItemInner Pagination-listItemInner--link" @click=goToPage(prevPage)>{{ prevPage }}</button>
    </li>
    <li class="Pagination-listItem is-current">
      <button class="Pagination-listItemInner Pagination-listItemInner--link" @click=goToPage(page)>{{ page }}</button>
    </li>
    <li class="Pagination-listItem" v-if="showNPositionNext(1)">
      <button class="Pagination-listItemInner Pagination-listItemInner--link" @click=goToPage(nextPage)>{{ nextPage }}</button>
    </li>
    <li class="Pagination-listItem" v-if="showNPositionNext(2)">
      <button class="Pagination-listItemInner Pagination-listItemInner--more">&hellip;</button>
    </li>
    <li class="Pagination-listItem" v-if="showLast">
      <button class="Pagination-listItemInner Pagination-listItemInner--link" @click=goToPage(numPages)>{{ numPages }}</button>
    </li>
  </ul>
</template>

<script>

export default {
  name: 'Pagination',
  props: {
    page: Number,
    numPages: Number
  },
  computed: {
    prevPage () {
      return this.page - 1;
    },
    nextPage () {
      return this.page + 1;
    },
    showFirst () {
      return this.page !== 1;
    },
    showLast () {
      return this.page !== this.numPages;
    }
  },
  methods: {
    showNPositionPrev (n) {
      return (this.page - n) > 1;
    },
    showNPositionNext (n) {
      return (this.page + n) < this.numPages;
    },
    goToPage (page) {
      this.$emit('pageChange', page);
    }
  }
};
</script>

<style scoped lang="scss">
@import 'stylesheets/new-dashboard/variables';

.Pagination-list {
  display: flex;
  align-items: center;
  justify-content: center;
}

.Pagination-listItem {
  border: 1px $grey solid;
  border-right-width: 0;
}

.Pagination-listItemInner {
  display: inline-block;
  padding: 15px 20px;
  border: none;
  outline: none;
  background: $white;
  color: $primary-color;
}

.Pagination-listItemInner--more {
  background-color: $white;
  color: $text-color;
  cursor: default;
}

.Pagination-listItem:first-child {
  border-radius: 2px 0 0 2px;

  .Pagination-listItemInner {
    border-radius: 2px 0 0 2px;
  }
}

.Pagination-listItem:last-child {
  border-right-width: 1px;
  border-radius: 0 2px 2px 0;

  .Pagination-listItemInner {
    border-radius: 0 2px 2px 0;
  }
}

.Pagination-listItem.is-current,
.Pagination-listItem.is-current .Pagination-listItemInner--link,
.Pagination-listItem.is-current .Pagination-listItemInner--link:hover {
  color: $text-color;
  background-color: $softblue;
  text-decoration: none;
  cursor: default;
}
</style>
