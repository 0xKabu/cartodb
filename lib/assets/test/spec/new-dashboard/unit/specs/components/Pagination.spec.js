import { shallowMount } from '@vue/test-utils';
import Pagination from 'new-dashboard/components/Pagination';

// Vue imports
// import VueRouter from 'vue-router';
// import Vuex from 'vuex';

// const localVue = createLocalVue();
// localVue.use(Vuex);
// localVue.use(VueRouter);

describe('Pagination.vue', () => {
  it('should render pagination for 6 pages from page number 1 with 1 ellipsis', () => {
    const pageCase = {
      page: 1,
      numPages: 6
    };
    const pagination = shallowMount(Pagination, {
      propsData: pageCase
    });

    expect(pagination).toMatchSnapshot();
  });
  it('should render pagination for 2 pages from page number 2', () => {
    const pageCase = {
      page: 2,
      numPages: 2
    };
    const pagination = shallowMount(Pagination, {
      propsData: pageCase
    });

    expect(pagination).toMatchSnapshot();
  });
  it('should render pagination for 9 pages from page number 5 with 2 ellipsis', () => {
    const pageCase = {
      page: 5,
      numPages: 9
    };
    const pagination = shallowMount(Pagination, {
      propsData: pageCase
    });

    expect(pagination).toMatchSnapshot();
  });
  it('should render pagination for 8 pages from page number 5 with 1 ellipsis', () => {
    const pageCase = {
      page: 5,
      numPages: 8
    };
    const pagination = shallowMount(Pagination, {
      propsData: pageCase
    });

    expect(pagination).toMatchSnapshot();
  });
});
