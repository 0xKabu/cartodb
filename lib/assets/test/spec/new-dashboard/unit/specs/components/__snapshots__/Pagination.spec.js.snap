// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination.vue should render pagination for 2 pages from page number 2 1`] = `
<ul class="Pagination-list CDB-Text CDB-Size-medium">
  <li class="Pagination-listItem"><button class="Pagination-listItemInner Pagination-listItemInner--link">1</button></li>
  <!---->
  <!---->
  <li class="Pagination-listItem is-current"><button class="Pagination-listItemInner Pagination-listItemInner--link">2</button></li>
  <!---->
  <!---->
  <!---->
</ul>
`;

exports[`Pagination.vue should render pagination for 6 pages from page number 1 with 1 ellipsis 1`] = `
<ul class="Pagination-list CDB-Text CDB-Size-medium">
  <!---->
  <!---->
  <!---->
  <li class="Pagination-listItem is-current"><button class="Pagination-listItemInner Pagination-listItemInner--link">1</button></li>
  <li class="Pagination-listItem"><button class="Pagination-listItemInner Pagination-listItemInner--link">2</button></li>
  <li class="Pagination-listItem"><button class="Pagination-listItemInner Pagination-listItemInner--more">…</button></li>
  <li class="Pagination-listItem"><button class="Pagination-listItemInner Pagination-listItemInner--link">6</button></li>
</ul>
`;

exports[`Pagination.vue should render pagination for 8 pages from page number 5 with 1 ellipsis 1`] = `
<ul class="Pagination-list CDB-Text CDB-Size-medium">
  <li class="Pagination-listItem"><button class="Pagination-listItemInner Pagination-listItemInner--link">1</button></li>
  <li class="Pagination-listItem"><button class="Pagination-listItemInner Pagination-listItemInner--more">…</button></li>
  <li class="Pagination-listItem"><button class="Pagination-listItemInner Pagination-listItemInner--link">4</button></li>
  <li class="Pagination-listItem is-current"><button class="Pagination-listItemInner Pagination-listItemInner--link">5</button></li>
  <li class="Pagination-listItem"><button class="Pagination-listItemInner Pagination-listItemInner--link">6</button></li>
  <li class="Pagination-listItem"><button class="Pagination-listItemInner Pagination-listItemInner--more">…</button></li>
  <li class="Pagination-listItem"><button class="Pagination-listItemInner Pagination-listItemInner--link">8</button></li>
</ul>
`;

exports[`Pagination.vue should render pagination for 9 pages from page number 5 with 2 ellipsis 1`] = `
<ul class="Pagination-list CDB-Text CDB-Size-medium">
  <li class="Pagination-listItem"><button class="Pagination-listItemInner Pagination-listItemInner--link">1</button></li>
  <li class="Pagination-listItem"><button class="Pagination-listItemInner Pagination-listItemInner--more">…</button></li>
  <li class="Pagination-listItem"><button class="Pagination-listItemInner Pagination-listItemInner--link">4</button></li>
  <li class="Pagination-listItem is-current"><button class="Pagination-listItemInner Pagination-listItemInner--link">5</button></li>
  <li class="Pagination-listItem"><button class="Pagination-listItemInner Pagination-listItemInner--link">6</button></li>
  <li class="Pagination-listItem"><button class="Pagination-listItemInner Pagination-listItemInner--more">…</button></li>
  <li class="Pagination-listItem"><button class="Pagination-listItemInner Pagination-listItemInner--link">9</button></li>
</ul>
`;
